lib/appsignal/sample_data.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+module Appsignal
+  # @api private
+  class SampleData
+    def initialize(key, accepted_type = nil)
+      @key = key
+      @accepted_type = accepted_type
+      @blocks = []
+    end
+
+    def add(data = nil, &block)
+      if block_given?
+        @blocks << block
+      elsif accepted_type?(data)
+        @blocks << data
```

Comment:
When merging this with #1221, I think we'll need to implement a `.dup` method for this SampleData class that dups the blocks array -- otherwise, I think that when the blocks are applied in the duplicate transactions, they will alter the blocks in the original transactions' sample data.

Reply:
Let's merge #1221 first and see what changes this PR requires before merging.